wpilibc/src/main/native/cpp/commands/FunctionalCommand.cpp

```diff
@@ -0,0 +1,37 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "frc/commands/FunctionalCommand.h"
+
+using namespace frc;
+
+FunctionalCommand::FunctionalCommand(std::function<void()> action)
+    : InstantCommand(), m_action(action) {}
+
+FunctionalCommand::FunctionalCommand(Subsystem& subsystem,
+                                     std::function<void()> action)
+    : InstantCommand(), m_action(action) {
+  Requires(&subsystem);
+}
+
+FunctionalCommand::FunctionalCommand(const wpi::Twine& name,
+                                     std::function<void()> action)
+    : InstantCommand(name), m_action(action) {}
+
+FunctionalCommand::FunctionalCommand(const wpi::Twine& name,
+                                     Subsystem& subsystem,
+                                     std::function<void()> action)
+    : InstantCommand(name), m_action(action) {
+  Requires(&subsystem);
+}
+
+void frc::FunctionalCommand::_Initialize() {
+  if (m_action) {
```

Comment:
If the user is required to pass a std::function in the constructor, it's exceedingly likely that m_action isn't nullptr. The only way it could be nullptr is if the user explicitly passed in `nullptr` as the argument. That is definitely not normal usage or even a sane thing to do. As such, you could just remove the if statement here.